internal/manager/builder.go

```diff
@@ -0,0 +1,128 @@
+package manager
```

Comment:
BTW, why is this internal?
Do we want to forbid users from launching our manager?
I think they still can do it by leveraging pkg/controller code. I would give them a proper way instead of forcing then at a lower level.

Reply:
@josvazg  `internal` usually contains code that is not supposed to be imported by other projects

Reply:
Precisely, I am not sure we want to prevent consumers from being able to use Setup when they still can go and use pkg code directly, at a lower level and with more pitfalls for them. I'd rather let them use what we use.